### PR TITLE
[jasmine] Add new 'toBePending' async matcher

### PR DESCRIPTION
**Pull request by AaronSherwoodForest**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jasmine/jasmine/pull/1808/files
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Hello! This is a partial update of jasmine to version 3.6.0. I saw this issue: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/46368 in which a contributor suggested adding only some of the new matchers as a valid option, and so I've added only one matcher that I needed. As such, I wasn't sure how to act on the checklist item above, "If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header."

If this pull request won't be accepted without applying all the necessary changes to bring the type definition up to the new version, I understand, and sorry for wasting your time!

### CODE REVIEW

**Comment by chivesrs** [comment]
This looks fine to me typings wise.

Just to verify runtime behavior, can you create a plnkr to verify this behavior? You can look at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45049#issuecomment-647077720 for an example and update the `<script>` tag it references to the right Jasmine version.

**Comment by SnailCoil** [comment]
Sure thing @chivesrs, https://plnkr.co/edit/k7uOcByK3AkP7eMK

**Comment by chivesrs** [comment]
Thank you. Approving on my end.

You might need to wait for a repo owner as well given Jasmine is marked as critical.

**Review by chivesrs — APPROVED** [review]

**Review by devoto13 — APPROVED** [review]

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -676,6 +676,12 @@ declare namespace jasmine {
     }
 
     interface AsyncMatchers<T, U> {
+        /**
+         * Expect a promise to be pending, ie. the promise is neither resolved nor rejected.
```
Comment (NothingEverHappens) [inline types/jasmine/index.d.ts:680]: uber nit: i.e.

**Review by NothingEverHappens — APPROVED** [review]

File: types/jasmine/index.d.ts (hunk 1 above)
Comment (SnailCoil) [reply types/jasmine/index.d.ts:680]: good catch! this typo exists verbatim in the source library, too. this is the first time I've contributed to DefinitelyTyped, do you know if there is a convention on fixing comments that are copy pasted from the library?

Comment (NothingEverHappens) [reply types/jasmine/index.d.ts:680]: Not that I know of, but should be simple to fix it in the lib as well?

**Review by NothingEverHappens — APPROVED** [review]
LGTM

**Review by chivesrs — APPROVED** [review]

**Review by devoto13 — APPROVED** [review]

**Review by kirjs — APPROVED** [review]

**Review by NothingEverHappens — APPROVED** [review]

**Review by chivesrs — APPROVED** [review]

**Comment by SnailCoil** [comment]
Ready to merge

**Pull request merged**